aioresponses/core.py

```diff
@@ -298,19 +348,86 @@ def add(self, url: 'Union[URL, str, Pattern]', method: str = hdrs.METH_GET,
 
     async def match(
             self, method: str, url: URL, **kwargs: Dict
-    ) -> Optional['ClientResponse']:
+    ) -> (Optional['ClientResponse'], Optional['RequestMatch']):
         for i, matcher in enumerate(self._matches):
             if matcher.match(method, url):
                 response = await matcher.build_response(url, **kwargs)
                 break
         else:
-            return None
+            return None, None
 
         if matcher.repeat is False:
             del self._matches[i]
         if isinstance(response, Exception):
             raise response
-        return response
+        return response, matcher
+
+    def _call_matcher(self, _call):
+        """
+        Given a call (or simply an (args, kwargs) tuple), return a
+        comparison key suitable for matching with other calls.
+        This is a best effort method which relies on the spec's signature,
+        if available, or falls back on the arguments themselves.
+        """
+
+        if isinstance(_call, RequestMatch):
+            return _call
+        if len(_call) == 2:
+            args, kwargs = _call
+            return RequestMatch(*args, **kwargs)
+        else:
+            raise NotImplementedError
+
+    def assert_not_called(self):
+        """assert that the mock was never called.
+        """
+        if len(self._calls) != 0:
+            msg = ("Expected '%s' to not have been called. Called %s times."
+                   % (self.__class__,
+                      len(self._calls)))
+            raise AssertionError(msg)
+
+    def assert_called(self):
+        """assert that the mock was never called.
+        """
+        if len(self._calls) == 0:
+            msg = ("Expected '%s' to not have been called. Called %s times."
+                   % (self.__class__,
+                      len(self._calls)))
+            raise AssertionError(msg)
+
+    def assert_called_once(self):
+        """assert that the mock was called only once.
+        """
+        if not len(self._calls) == 1:
+            msg = ("Expected '%s' to have been called once. Called %s times."
+                   % (self.__class__ or 'mock',
+                      len(self._calls)))
+
+            raise AssertionError(msg)
+
+    def assert_called_with(self, *args, **kwargs):
+        """assert that the mock was called with the specified arguments.
+        Raises an AssertionError if the args and keyword args passed in are
+        different to the last call to the mock."""
+        self.assert_called()
+
+        expected = self._call_matcher((args, kwargs))
+        actual = self._call_matcher(self._call_args)
```

Comment:
Scenario. 
User mocks two or more requests (`http://example.com` and `http://httpbin.org/`)
What I would expected is to successfully call 
`assert_called_with('http://example.com')` 
and  
`assert_called_with('http://httpbin.org/')`

With current implementation this is not possible :/

Reply:
I've added list of calls `_request_match_list` and method `assert_any_call` that checks if any call like that was made. You can find a test for this functionality in `test_assert_any_call_success` :ok_hand: